core/arch/arm/kernel/boot.c

```diff
@@ -374,12 +374,28 @@ static void init_asan(void)
 /* Called from entry_a64.S only when MEMTAG is configured */
 void boot_init_memtag(void)
 {
-	paddr_t base = 0;
-	paddr_size_t size = 0;
-
 	memtag_init_ops(feat_mte_implemented());
-	core_mmu_get_secure_memory(&base, &size);
-	memtag_set_tags((void *)(vaddr_t)base, size, 0);
+}
+
+/* Called from entry_a64.S only when MEMTAG is configured */
+void boot_clear_memtag(void)
+{
+	enum teecore_memtypes mtypes[] = {
+		MEM_AREA_TEE_RAM, MEM_AREA_TEE_RAM_RW, MEM_AREA_NEX_RAM_RW,
+		MEM_AREA_TEE_ASAN, MEM_AREA_TA_RAM
```

Comment:
Maybe a stupid question but yhy are `MEM_AREA_TEE_RAM_RW` & `MEM_AREA_NEX_RAM_RW` considered but not the other tagged memory types:
`MEM_AREA_TEE_RAM_RX`
`MEM_AREA_INIT_RAM_RX`
`MEM_AREA_IDENTITY_MAP_RX`
`MEM_AREA_TEE_RAM_RO`
`MEM_AREA_INIT_RAM_RO`
`MEM_AREA_NEX_RAM_RO`

Reply:
We can't write to tags in read-only memory.

Reply:
Ok thanks, indeed you mention that in the commit message.
From [core_mmu.c](https://elixir.bootlin.com/op-tee/4.1.0/source/core/mm/core_mmu.c#L748) : `MEM_AREA_NEX_RAM_RO` is writable during the initialization:
```c
case MEM_AREA_NEX_RAM_RO: /* This has to be r/w during init runtime */
```
Should it be considered here?

Reply:
I guess it doesn't harm to add that too.